codec/src/main/java/io/netty/handler/codec/DelimiterBasedFrameDecoder.java

```diff
@@ -268,13 +268,13 @@ protected Object decode(ChannelHandlerContext ctx, ByteBuf buffer) throws Except
             }
 
             if (stripDelimiter) {
-                frame = buffer.readSlice(minFrameLength);
+                frame = buffer.readRetainedSlice(minFrameLength);
                 buffer.skipBytes(minDelimLength);
```

Comment:
I don't think so. buffer.readableBytes() must be >= minFrameLength + minDelimLength.